AdamantTests/Modules/Wallets/DashWalletServiceIntegrationTests.swift

```diff
@@ -0,0 +1,135 @@
+        // given
+        sut.setWalletForTests(try makeWallet())
+        let data = Constants.unspentTranscationsData
+        await apiCoreMock.isolated { mock in
+            mock.stubbedSendRequestBasicGenericResult = APIResponseModel(result: .success(data), data: data, code: 200)
+        }
+        
+        // when 1
+        let result = await Result(catchingAsync: {
+            try await self.sut.createTransaction(
+                recipient: "Xp6kFbogHMD4QRBDLQdqRp5zUgzmfj1KPn",
+                amount: 0.01,
+                fee: 0.0000001,
+                comment: nil
+            )
+        })
+        
+        // then 1
+        let transaction = try XCTUnwrap(result.value)
+        XCTAssertEqual(transaction.serialized().hex, Constants.expectedTransactionHex)
+        XCTAssertEqual(transaction.txID, Constants.expectedTransactionID)
+        
+        // given 2
+        let txData = Constants.sendTransactionResponseData
+        await apiCoreMock.isolated { mock in
+            mock.stubbedSendRequestBasicGenericResult = APIResponseModel(result: .success(txData), data: txData, code: 200)
+        }
+        
+        // when 2
+        let result2 = await Result {
+            try await self.sut.sendTransaction(transaction)
+        }
+        // then 3
+        XCTAssertNil(result2.error)
+        await apiCoreMock.isolated { mock in
+            XCTAssertEqual(mock.invokedSendRequestBasicGenericCount, 2)
+        }
+    }
+}
+
+// MARK: Private
+
+private extension DashWalletServiceIntegrationTests {
+    func makeWallet() throws -> DashWallet {
+        let privateKeyData = Constants.passphrase
+            .data(using: .utf8)!
+            .sha256()
+        let privateKey = PrivateKey(
+            data: privateKeyData,
+            network: DashMainnet(),
+            isPublicKeyCompressed: true
+        )
+        return try DashWallet(
+            unicId: "DASHDASH",
+            privateKey: privateKey,
+            addressConverter: AddressConverterFactory().make(network: DashMainnet())
+        )
+    }
+}
+
+private enum Constants {
+    
+    static let passphrase = "village lunch say patrol glow first hurt shiver name method dolphin dead"
+    
+    static let expectedTransactionID = "d4cf3fde45d0e7ba855db9621bdc6da091856011d86f199bebd1937f7b63020a"
+    
+    static let expectedTransactionHex = "0100000001721f0c2437124acb8a20fea5af60908057b086118b240119adcb39c890b4c3a2010000006a473044022002e19bc62748ca3f34a6e5f1aeab31bb3dc43997792d9551e9b8c51a094abbae02200e3c9bde7c60326ef6984045a81304a9915e8fbce96de01813fe89886ef26453012102cd3dcbdfc1b77e54b3a8f273310806ab56b0c2463c2f1677c7694a89a713e0d0ffffffff0240420f00000000001976a914931ef5cbdad28723ba9596de5da1145ae969a71888acb695a905000000001976a91457f6f900ac7a7e3ccab712326cd7b85638fc15a888ac00000000"
+    
+    static let unspentTranscationsData = unspentTransactionsRawJSON.data(using: .utf8)!
+    
+    static let unspentTransactionsRawJSON: String = """
```

Comment:
We might want to use actual JSON files, fixtures that is, in order to have a more unified way of editing and searching for them when writing more and more tests in future.
What do you think @ChrisBenua?

Reply:
Done